### PR TITLE
Unify the the Printing Interface

### PR DESCRIPTION
**Pull request by hazen**
The C printing interface had developed over the months and evolved.  There were a couple of different flavors; some were very cumbersome.  Here we reign in the differences and create a cleaner, public interface.  This addresses Issue #40.  Also incorporates some ideas from PR #44 from @folays.

### CODE REVIEW

File: src/riak_print.c [hunk 1]
```diff
@@ -23,190 +23,151 @@
 #include <time.h>
```
Comment (bookshelfdave) [inline src/riak_print.c:23]: the header of riak_print.c lists:
`* utils.c: Riak C Client Utilities`

File: src/riak_print.c [hunk 2]
```diff
@@ -23,190 +23,151 @@
...
+riak_print_label_binary_hex(riak_print_state *state,
+                            char             *name,
+                            riak_binary      *value) {
+    riak_int32_t wrote = riak_print(state, "%s: ", name);
+    if (state->len > 0) {
+        wrote += riak_print_binary_hex(state, value);
     }
+    wrote += riak_print(state, "%s", "\n");
+
+    return wrote;
```
Comment (bookshelfdave) [inline src/riak_print.c:123]: multiple returns from this function (see below)

File: src/messages/riak_delete.c [hunk 3]
```diff
@@ -127,34 +126,33 @@
...
-        riak_print_bool("Sloppy Quorum", opt->sloppy_quorum, &target, &len, &total);
+        wrote += riak_print_label_bool(state, "Sloppy Quorum", opt->sloppy_quorum);
     }
     if (opt->has_n_val) {
-        riak_print_int("N Values", opt->n_val, &target, &len, &total);
+        wrote += riak_print_label_int(state, "N Values", opt->n_val);
     }
 
-    return total;
+    return wrote;
```
Comment (bookshelfdave) [inline src/messages/riak_delete.c:155]: since state is tracked in `riak_print_state`, do you need to return `wrote` from the _print_ functions?

**Comment by bookshelfdave** [comment]
seems fine, +1 if you fix that dup return()

File: test/cunit/test_serverinfo.c [hunk 4]
```diff
@@ -154,9 +156,11 @@
...
                          void                     *ptr) {
     test_async_connection *conn = (test_async_connection*)ptr;
-    char buffer[1024];
-    riak_serverinfo_response_print(response, buffer, sizeof(buffer));
-    fprintf(stderr, "ASYNCHRONOUS %s", buffer);
+    char output[1024];
+    riak_print_state print_state;
+    riak_print_init(&print_state, output, sizeof(output));
+    riak_serverinfo_response_print(&print_state, response);
+    fprintf(stderr, "ASYNCHRONOUS %s", output);
```
Comment (bookshelfdave) [inline test/cunit/test_serverinfo.c:163]: :crocodile: ASYNCHRONOUS :crocodile:

Comment (hazen) [inline test/cunit/test_serverinfo.c:163]: Here I though it was a typo, but it appears simply to be code gremlins :smiling_imp:

File: src/messages/riak_delete.c (hunk 3 above)
Comment (hazen) [inline src/messages/riak_delete.c:155]: It is a fair point.  Some had it, others didn't.  I was just being consistent, but you are right. Probably don't need it.

**Pull request merged**